messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -437,6 +436,7 @@ private void processBatch(Segment segment, BlockingStream<TrackedEventMessage<?>
                     return;
                 }
             } else {
+                checkSegmentCaughtUp(segment, eventStream);
```

Comment:
There is a check after the entire `if` block, so I don't see why there should be one here?

Reply:
Thought was to allow validation if the `caughtUp` flag in the scenario the `BlockingStream` would not return at all. As the `else` reaches a `return` statement, we would thus miss updating the `caughtUp` flag in case the event stream was actually empty.